docs/content/contributing/bicep/bicep-contribution-flow/_index.md

```diff
@@ -100,55 +107,122 @@ For more examples, see the below script's parameters section.
+
+In those cases, for the first PR adding such modules to the public registry, we recommend the author to reach out to AVM maintainers or, alternatively, to [create a CI environment GitHub issue](https://github.com/Azure/bicep-registry-modules/issues/new?template=avm_ci_environment_issue.yml) in BRM, specifying the additional prerequisites. This is to make sure that the required additional roles get assigned in the upstream CI environment before the corresponding module PR gets merged.
+
+{{< /hint >}}
+
+2. Configure a federated identity credential on a user-assigned managed identity to trust tokens issued by GitHub Actions to your GitHub repository.
+    - In the Microsoft Entra admin center, navigate to the user-assigned managed identity you created. Under `Settings` in the left nav bar, select `Federated credentials` and then `Add Credential`.
+      <img src="../../../img/bicep-ci/msiOIDCAddFederatedIdentity_01.png" alt="OIDC federated credentials" width=60%>
+    - In the Federated credential scenario dropdown box, select `GitHub Actions deploying Azure resources`
+    <img src="../../../img/bicep-ci/msiOIDCAddFederatedIdentity_02.png" alt="OIDC scenario" width=80%>
+    - For Organization, specify your GitHub organization name, for Repository specify the value `bicep-registry-modules`.
+    - For Entity type, select `Environment` and specify the value `avm-validation`.
+    - Add a Name for the federated credential, for example, `avm-gh-env-validation`.
+    - The Issuer, Audiences, and Subject identifier fields autopopulate based on the values you entered.
+    - Select `Add` to configure the federated credential.
+    <img src="../../../img/bicep-ci/msiOIDCAddFederatedIdentity_03.png" alt="OIDC Add" width=80%>
+    - You might find the following links useful:
+      - [Configure a federated identity credential on a user-assigned managed identity](https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-create-trust-user-assigned-managed-identity)
+3. Note down the following pieces of information
+    - Client ID
+    - Tenant ID
+    - Subscription ID
+    - Parent Management Group ID
+
+    <img src="../../../img/bicep-ci/msiOIDCInfo.png" alt="OIDC Add" width=70%>
+
+Additional references:
+- [Configure a federated identity credential](https://learn.microsoft.com/en-us/azure/developer/github/connect-from-azure-openid-connect#prerequisites)
+- [Azure login GitHub action - Login with OIDC](https://github.com/Azure/login?tab=readme-ov-file#login-with-openid-connect-oidc-recommended)
+
+{{< /expand >}}
+
+{{< expand "➕ Option 2 [Deprecated]: Configure Service Principal + Secret" "expand/collapse" >}}
+
+1. Create a new or leverage an existing Service Principal with at least `Contributor` & `User Access Administrator` permissions on the Management-Group/Subscription you want to test the modules in. You might find the following links useful:
```

Comment:
Same ask as in [this comment](https://github.com/Azure/Azure-Verified-Modules/pull/1773/files#r1898999613), whether we should recommend a different role than `User Access Administrator`